src/Microsoft.AspNetCore.Razor/Compilation/TagHelpers/TagHelperRequiredAttributeDescriptor.cs

```diff
@@ -0,0 +1,90 @@
+{
+    /// <summary>
+    /// A metadata class describing a required tag helper attribute.
+    /// </summary>
+    public class TagHelperRequiredAttributeDescriptor
+    {
+        /// <summary>
+        /// Supported CSS value operators.
+        /// </summary>
+        public static readonly char[] SupportedCSSValueOperators = { '=', '^', '$' };
+
+        /// <summary>
+        /// The HTML attribute name.
+        /// </summary>
+        public string Name { get; set; }
+
+        /// <summary>
+        /// The HTML attribute selector value. If <see cref="IsCssSelector"/> is not <c>true</c>, this field is
+        /// ignored.
+        /// </summary>
+        public string Value { get; set; }
+
+        /// <summary>
+        /// An operator that modifies how a required attribute is applied to an HTML attribute value or name.
+        /// </summary>
+        public char Operator { get; set; }
+
+        /// <summary>
+        /// Indicates if the <see cref="TagHelperRequiredAttributeDescriptor"/> represents a CSS selector.
+        /// </summary>
+        public bool IsCssSelector { get; set; }
+
+        /// <summary>
+        /// Determines if the current <see cref="TagHelperRequiredAttributeDescriptor"/> matches the given
+        /// <paramref name="attributeName"/> and <paramref name="attributeValue"/>.
+        /// </summary>
+        /// <param name="attributeName">An HTML attribute name.</param>
+        /// <param name="attributeValue">An HTML attribute value.</param>
+        /// <returns></returns>
+        public bool IsMatch(string attributeName, string attributeValue)
+        {
+            if (IsCssSelector)
+            {
+                if (!string.Equals(Name, attributeName, StringComparison.OrdinalIgnoreCase))
+                {
+                    return false;
+                }
+
+                var valueMatches = false;
+                switch (Operator)
+                {
+                    case '^': // Value starts with
+                        valueMatches = attributeValue.StartsWith(Value, StringComparison.Ordinal);
+                        break;
+                    case '$': // Value ends with
+                        valueMatches = attributeValue.EndsWith(Value, StringComparison.Ordinal);
+                        break;
+                    case '=': // Value equals
+                        valueMatches = string.Equals(attributeValue, Value, StringComparison.Ordinal);
+                        break;
+                    case '\0': // No value selector, force true because at least the attribute name matched.
+                        valueMatches = true;
+                        break;
+                    default:
+                        Debug.Assert(false, "Unknown operator.");
+                        break;
+                }
+
+                return valueMatches;
+            }
+            else if (Operator == '*')
+            {
+                return attributeName.Length != Name.Length &
```

Comment:
`&` was written instead of `&&` due to a failure of my hands not continuing to type a second `&`. They shall be punished and an extra `&` will be added.